forum_service.py

```diff
@@ -0,0 +1,114 @@
+        self.db_connection.commit()
+        cursor.close()
+
+    def untrack_thread(self, thread_id):
+        """this method will untrack a thread"""
+        cursor = self.db_connection.cursor()
+        cursor.execute('UPDATE threads SET active=0 WHERE thread_id = ?', (thread_id,))
+        self.db_connection.commit()
+        cursor.close()
+
+    def track_new_thread(self, url):
+        """
+            starts tracking a new thread by adding some info to the threads table. When a new thread is tracked, it automatically
+            sets the values (last_page_num, last_post_num) to (1,1). 
+        """
+        cursor = self.db_connection.cursor()
+        cursor.execute('INSERT INTO threads (url, active, last_page_num, last_post_num) VALUES (?, ?)', (url, 1, 1, 1))
+        self.db_connection.commit()
+        cursor.close()
+        #should maybe add a boolean return 
+
+    def insert_raw_posts(self, raw_data):
+        """
+        will insert raw post data into the raw_posts table. We should aim to deprecate this
+        """
+        cursor = self.db_connection.cursor()
+        for post in raw_data:
+            cursor.execute('INSERT INTO raw_posts VALUES (?, ?, ?, ?)', (post[1], post[2], post[0], str(dt.now())))
+        self.db_connection.commit()
+        cursor.close()
+
+    def insert_price_reports(self):
+        """
+        This will eventually replace the insert_raw_posts methods by storing cleaned information
+        """
+        pass
+
+    def fetch_price_reports(self, item_name):
+        """
+        this will fetch all price reports for a specific item. NOTE: This may be to vague to be useful
+        """
+        cursor = self.db_connection.cursor()
+        #cursor.execute('SELECT * FROM price_reports WHERE full_name = ?', (item_name,))
+        cursor.execute('SELECT pr.* FROM price_reports pr, item_lookup il WHERE pr.item_id = il.item_id AND il.short_name = ?', (item_name,))
+        price_reports = cursor.fetchall()
+        return price_reports
+
+    ##################
+    # Scraping Methods
+    ##################
+    def scrape_active_threads(self):
+        """
+        this method will take all active threads, scrape them starting from last_page until max_page
+        it will then insert raw post data into the raw_posts database <---replace this with cleaned data
+        finally it updates the threads table with the last page/post information
+        """
+        active_threads = self.fetch_active_threads()
+        for thread in active_threads:
+            url, last_page, last_post, thread_id = thread
+            scraper = RSScraper(url)
+            for page_index in range(last_page, last_page+5):
+                data = scraper.scrape_page(page_index)
+                self.insert_raw_posts(data)
+                self.update_tracked_threads(thread_id, scraper.current_page, scraper.current_post)
+
+
+
+    def get_all_price_posts(self, url):
```

Comment:
these get methods should take a callback function that gets called when the posts have all been scraped -- something like:
```suggestion
    def get_all_price_posts(self, url, completion):
        ... do scraping stuff ...
        completion(scrapedPosts)
```

Reply:
We can do this in a future ticket. Can you create a ticket to do this? @binzabinza